src/supertux/menu/sorted_contrib_menu.cpp

```diff
@@ -38,10 +41,78 @@ SortedContribMenu::SortedContribMenu(std::vector<std::unique_ptr<World>>& worlds
+          title_str = str(boost::format(_("[%s] *NEW*")) % worlds[i]->get_title());
+        }
+        else
+        {
+          /* This is translatable since RTL languages may prefer to put the progress
+             info to the left of the title */
+          title_str = str(boost::format(_("[%s] (%u/%u; %u%%)")) % worlds[i]->get_title() %
+                          solved_count % level_count % (100 * solved_count / level_count));
+        }
+      }
       else
-        title_str = worlds[i]->get_title();
+      {
+        uint32_t island_level_count = 0, island_solved_count = 0,
+                 world_level_count = 0,  world_solved_count = 0;
+        auto wm_filename = savegame->get_player_status().last_worldmap;
+        if (wm_filename.empty())
+          wm_filename = "/" + worlds[i]->get_worldmap_filename();
+        const auto& state = savegame->get_worldmap_state(wm_filename);
+        for (const auto& level_state : state.level_states)
+        {
+          if (level_state.filename.empty()) continue;
+          if (level_state.solved) ++island_solved_count;
+          ++island_level_count;
+        }
+        const auto levelset = std::unique_ptr<Levelset>(new Levelset(worlds[i]->get_basedir(), true));
+        world_level_count = levelset->get_num_levels();
+        const auto world_list = savegame->get_worldmaps();
+        for (const auto& world : world_list)
+        {
+          const auto& world_state = savegame->get_worldmap_state(world);
+          for (const auto& level_state : world_state.level_states)
+          {
+            if (level_state.filename.empty()) continue;
+            if (level_state.solved) ++world_solved_count;
+          }
+        }
+        if (!island_level_count && !world_solved_count)
+          title_str = str(boost::format(_("%s *NEW*")) % worlds[i]->get_title());
+        else
+        {
+          const auto wm_title = savegame->get_player_status().last_worldmap_title;
+          if (island_level_count == world_level_count)
+          {
+            /* This is translatable since RTL languages may prefer to put the progress
+               info to the left of the title */
+            title_str = str(boost::format(_("%s (%u/%u; %u%%)")) % worlds[i]->get_title() %
+                            island_solved_count % island_level_count % (100 * island_solved_count / island_level_count));
+          }
+          else
+          {
+            /* This is translatable since RTL languages may prefer to put the progress
+               info to the left of the title */
+            title_str = str(boost::format(_("%s (%u/%u; %u%%) - %s (%u/%u; %u%%)")) % worlds[i]->get_title() % 
+                            world_solved_count % world_level_count % (100 * world_solved_count / world_level_count) %
+                            wm_title % island_solved_count % island_level_count %
+                            (island_level_count ? (100 * island_solved_count / island_level_count) : 100));
+          }
```

Comment:
Is there a reason to duplicate the count for a world and for a worldmap? Isn't the world's stats enough?

Reply:
I originally thought it may be enough the other way round :D

I think having both counts is good for bigger worlds with more than one worldmap, like 
![image](https://user-images.githubusercontent.com/14074789/147924102-3a899104-823d-4434-8d49-414b03e9863a.png)
For worlds which only have one worldmap (ie the level count in the worldmap and the world are identical), only one will show.